src/main/java/com/autotune/common/utils/ExponentialBackOff.java

```diff
@@ -0,0 +1,190 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat, IBM Corporation and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.autotune.common.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * This builder class helps in retrying specific task using exponential backoff approach.
+ */
+public final class ExponentialBackOff {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExponentialBackOff.class);
+    // Maximum total time in Millis to complete exponential backoff approach. Default set to 10sec
+    public static long defaultMaxElapsedTimeMillis = 10 * 1000;   //1o Seconds
```

Comment:
The Default values can be placed in `AutotuneConstants` with a suitable variable names to keep it consistent.